src/servers/src/mysql/handler.rs

```diff
@@ -253,7 +277,16 @@ impl<W: AsyncWrite + Send + Sync + Unpin> AsyncMysqlShim<W> for MysqlInstanceShi
             let user_id = Identity::UserId(&username, addr.as_deref());
 
             let password = match auth_plugin {
-                "mysql_native_password" => Password::MysqlNativePassword(auth_data, salt),
+                MYSQL_NATIVE_PASSWORD => Password::MysqlNativePassword(auth_data, salt),
+                MYSQL_CLEAR_PASSWORD => {
+                    if auth_data.is_empty() {
+                        return false;
+                    }
+                    // The raw bytes received are represented in C-like string, ended in '\0'.
```

Comment:
I think we should add an assertion to ensure that it's ended with `\0`. To make sure the behavior has not changed.